src/main/modules/clef.js

```diff
@@ -0,0 +1,211 @@
+import { Notification, dialog } from 'electron';
+import { spawn } from 'cross-spawn';
+import log from 'electron-log';
+import path from 'path';
+import jsonrpc from 'jsonrpc-lite';
+import StringDecoder from 'string_decoder';
+import { validateTx } from '../../lib/validators';
+import { note } from './util';
+
+const decoder = new StringDecoder.StringDecoder('utf8');
+
+export function startClef(argv) {
+  let clef;
+  let clefpath;
+
+  if (argv.clefbin) {
+    clefpath = argv.clefbin;
+  } else {
+    clefpath = dialog.showOpenDialog({
+      title: 'Select Clef Binary',
+      filters: [
+        { name: 'Application', extensions: ['exe', 'bin', 'app'] },
+        { name: 'All Files', extensions: ['*'] }
+      ]
+    })[0];
+  }
+
+  try {
+    clef = spawn(clefpath, [
+      '--rpc',
+      '--4bytedb',
+      path.dirname(clefpath) + '/4byte.json',
+      '--ipcdisable',
+      '--stdio-ui'
+    ]);
+  } catch (error) {
+    log.error(error);
+    throw error;
+  }
+
+  notifyClefSha256(clefpath);
```

Comment:
It would be really nice if we could pop up a confirm-dialog. 
```
The clef binary has sha256-sum <foo>. 
[Cancel]                   [ Ok ]
```

Although in the long run, I'd like the UI to read a signed file containing official hashes.

Reply:
yeah for sure